assets/src/edit-story/components/canvas/eyedropperLayer.js

```diff
@@ -0,0 +1,259 @@
+import { rgba, readableColor } from 'polished';
+import { FULLBLEED_RATIO } from '@web-stories-wp/units';
+
+/**
+ * Internal dependencies
+ */
+import { useCanvas, useLayout } from '../../app';
+import { useFocusOut, useKeyDownEffect } from '../../../design-system';
+import { Layer, PageArea } from './layout';
+import getColorFromPixelData from './utils/getColorFromPixelData';
+
+const EyedropperBackground = styled(Layer)`
+  background: rgba(0, 0, 0, 0.7);
+  cursor: not-allowed;
+  z-index: 2;
+`;
+
+const DisplayPageArea = styled(PageArea)`
+  position: absolute;
+`;
+
+const EyedropperCanvas = styled.div`
+  background: transparent;
+  border-radius: 5px;
+  color: black;
+  position: absolute;
+  z-index: 2;
+  transform: translateZ(0);
+  top: 0;
+  left: 0;
+  bottom: 0;
+  right: 0;
+  cursor: default;
+`;
+
+const ColorInfo = styled.div`
+  position: absolute;
+  text-transform: uppercase;
+  transform: translateX(-50%);
+  padding: 9px;
+  min-width: 102px;
+  text-align: center;
+`;
+
+const MagnifierCanvas = styled.canvas`
+  border: 2px solid black;
+  border-radius: 50%;
+  transform: translate(calc(-50% + 4px), 20px);
+  background: ${({ theme }) => theme.colors.bg.primary};
+`;
+
+const Magnifier = styled.div`
+  position: absolute;
+  display: flex;
+  top: 0;
+  left: 0;
+  transform: translateY(-1000px);
+  pointer-events: none;
+`;
+
+function EyedropperLayer() {
+  const {
+    fullbleedContainer,
+    eyedropperActive,
+    eyedropperCallback,
+    eyedropperImg,
+    eyedropperPixelData,
+    setEyedropperActive,
+  } = useCanvas(
+    ({
+      state: {
+        fullbleedContainer,
+        eyedropperActive,
+        eyedropperCallback,
+        eyedropperImg,
+        eyedropperPixelData,
+      },
+      actions: { setEyedropperActive },
+    }) => ({
+      fullbleedContainer,
+      eyedropperActive,
+      eyedropperCallback,
+      eyedropperImg,
+      eyedropperPixelData,
+      setEyedropperActive,
+    })
+  );
+
+  const { pageWidth } = useLayout(({ state: { pageWidth } }) => ({
+    pageWidth,
+  }));
+
+  const fullHeight = pageWidth / FULLBLEED_RATIO;
+  const img = eyedropperImg;
+  const imgRef = useRef();
+  const magnifier = useRef();
+  const magnifierInfo = useRef();
+  const magnifierColor = useRef();
+
+  useEffect(() => {
+    const canvas = magnifier.current;
+    if (canvas) {
+      const ctx = canvas.getContext('2d');
+      ctx.imageSmoothingEnabled = false;
+      ctx.mozImageSmoothingEnabled = false;
+      ctx.webkitImageSmoothingEnabled = false;
+      ctx.msImageSmoothingEnabled = false;
+    }
+  }, [eyedropperImg]);
+
+  const eyedropperCanvas = useRef();
+
+  const closeEyedropper = () => setEyedropperActive(false);
+
+  useFocusOut(eyedropperCanvas, closeEyedropper, [eyedropperActive, img]);
+
+  useKeyDownEffect(document.body, 'esc', closeEyedropper);
+
+  if (!eyedropperActive || !img) {
+    return null;
+  }
+
+  const magnifierSize = 200; // We can use scroll wheel to control it?
```

Comment:
> // We can use scroll wheel to control it?

Do you mean to zoom in / out inside the magnifier? Probably can skip for now :)

We could perhaps move these above the component and use constants instead (`const MAGNIFIER_SIZE` etc.)